src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -357,6 +357,113 @@ public bool IsHomeSpace
+        /// Returns the Uuid property of the seralized Workspace
+        /// </summary>
+        [JsonIgnore]
+        internal string GraphId { get; set; }
+
+        /// <summary>
+        /// Returns the Json representation of the current graph
+        /// </summary>
+        [JsonIgnore]
+        internal JObject JsonRepresentation { get; set; }
+
+        /// <summary>
+        /// Returns the stringified representation of the connected nodes
+        /// </summary>
+        [JsonIgnore]
+        public string Checksum
+        {
+            get
+            {
+                List<string> nodeInfoConnections = new List<string>();
+                JObject jsonWorkspace = JsonRepresentation;
+                var nodes = jsonWorkspace["Nodes"];
+
+                List<string> nodeIds = new List<string>();
+                foreach (JObject node in nodes)
+                {
+                    var nodeProperties = node.Children<JProperty>();
+                    JProperty id = nodeProperties.FirstOrDefault(x => x.Name == "Id");
+                    nodeIds.Add(id.Value.ToString());
+                }
+
+                nodeIds.Sort();
+
+                foreach (string nodeId in nodeIds)
+                {
+                    List<string> outputIds = new List<string>();
+                    var node = jsonWorkspace["Nodes"].Where(t => t.Value<string>("Id") == nodeId).Select(t => t).FirstOrDefault();
+                    var outputsProperty = node.Children<JProperty>().FirstOrDefault(x => x.Name == "Outputs");
+                    var outputs = (JArray)outputsProperty.Value;
+                    int outputIndex = 1;
+
+                    foreach (JObject output in outputs)
+                    {
+                        var outputProperties = output.Children<JProperty>();
+                        JProperty outputId = outputProperties.FirstOrDefault(x => x.Name == "Id");
+                        outputIds.Add(outputId.Value.ToString());
+
+                        var connectorsProperty = jsonWorkspace["Connectors"].Where(t => t.Value<string>("Start") == outputId.Value.ToString());
+
+                        foreach (var connector in connectorsProperty)
+                        {
+                            var connectorProperties = connector.Children<JProperty>();
+                            JProperty endProperty = connectorProperties.FirstOrDefault(x => x.Name == "End");
+                            string inputId = (String)endProperty.Value;
+
+                            var outputConnectedNode = GetNodeByInputId(inputId, jsonWorkspace);
+                            nodeInfoConnections.Add(nodeId + "|[" + outputIndex.ToString() + "|" + outputConnectedNode.Item1 + "|" + outputConnectedNode.Item2.ToString() + "]");
+                        }
+                        outputIndex++;
+                    }
+                }
+                return nodeInfoConnections.Count > 0 ? string.Join(",", nodeInfoConnections) : string.Empty;
```

Comment:
Is the checksum a string of all nodes IDs and their connections in plain text?

Reply:
> Is the checksum a string of all nodes IDs and their connections in plain text?

yup, in order to be compared quickly with another version of the same graph and identify any nodes connections change